csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,49 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public void RemoveField(int fieldIndex)
+        {
+            if (fieldIndex < 0 || fieldIndex > _fields.Count)
+            {
+                throw new ArgumentException(nameof(fieldIndex), "Invalid fieldIndex");
+            }
+            _fieldsDictionary.Remove(_fields[fieldIndex].Name);
+            _fields.RemoveAt(fieldIndex);
+        }
+
+        public void AddField(Field newField, int fieldIndex)
+        {
+            newField = newField ?? throw new ArgumentNullException(nameof(newField));
+            if (fieldIndex < 0 || fieldIndex > _fields.Count)
+            {
+                throw new ArgumentException(nameof(fieldIndex), $"Invalid fieldIndex {fieldIndex} passed in to Schema.AddField");
+            }
+
+            List<Field> newFields = new List<Field>();
+            for (int ii = 0; ii < fieldIndex; ii++)
+            {
+                newFields.Add(_fields[ii]);
+            }
+            newFields.Add(newField);
```

Comment:
This should just be `_fields.Insert`.  And then that way the `_fields` property can remain `readonly`.